pkgs/by-name/da/daytona-bin/providers/docker/default.nix

```
@@ -0,0 +1,52 @@
+{ stdenvNoCC
+, lib
+, fetchurl
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "daytona-docker-provider-bin";
+  version = "0.2.1";
+
+  src =
+    let
+      urls = {
+        "x86_64-linux" = {
+          url = "https://download.daytona.io/daytona/providers/docker-provider/v${finalAttrs.version}/docker-provider-linux-amd64";
+          hash = "sha256-CaY3nJDe5Ym4YLwXFAPa2SCtTwjIWdfresp8HaNXedk=";
+        };
+        "x86_64-darwin" = {
+          url = "https://download.daytona.io/daytona/providers/docker-provider/v${finalAttrs.version}/docker-provider-darwin-amd64";
+          hash = "sha256-abbbDHsa5Nx4rP+lTDVFfNqnZi3uTMJZCTRAkLEhOlI=";
+        };
+        "aarch64-linux" = {
+          url = "https://download.daytona.io/daytona/providers/docker-provider/v${finalAttrs.version}/docker-provider-linux-arm64";
+          hash = "sha256-24UM6SHDGMOlN/uOYvurJVslfOIdNpKnVwSIPKzDCEc=";
+        };
+        "aarch64-darwin" = {
+          url = "https://download.daytona.io/daytona/providers/docker-provider/v${finalAttrs.version}/docker-provider-darwin-arm64";
+          hash = "sha256-v9H/JCebF7fYtSeix2RkBlXQ2S0xQDol5k30AGYjXNI=";
+        };
+      };
+    in
+    fetchurl urls."${stdenvNoCC.hostPlatform.system}";
+
+  dontUnpack = true;
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cp -r $src $out/bin/daytona-docker-provider
+    chmod +x $out/bin/daytona-docker-provider
```

IMO this is a bit clearer
```suggestion
    install -Dm755 $src $out/bin/daytona-docker-provider
```